code/obj/item/misc_weapons.dm

```diff
@@ -1269,9 +1269,24 @@ TYPEINFO(/obj/item/swords/katana)
 		mid1 = new/obj/itemspecialeffect/katana_dash/mid(src)
 		mid2 = new/obj/itemspecialeffect/katana_dash/mid(src)
 		end = new/obj/itemspecialeffect/katana_dash/end(src)
+		src.create_reagents(KATANA_REAGENT_CAPACITY)
 		src.setItemSpecial(/datum/item_special/katana_dash)
 		BLOCK_SETUP(BLOCK_SWORD)
 
+	afterattack(atom/target, mob/user)
+		..()
+		apply_coating(target, user)
+
+	proc/apply_coating(var/atom/target, var/mob/user)
+		if (target.is_open_container())
+			var/obj/item/reagent_containers/RC = target
+			if(RC.reagents.has_reagent("sakuride", 1))
+				if(RC.reagents.reagent_list.len > 1)
+					boutput(user, "<span class='alert'>This coating is impure!</span>")
+					return
+				RC.reagents.trans_to(src, min(KATANA_REAGENT_CAPACITY , src.reagents.maximum_volume - src.reagents.total_volume))
+				boutput(user, "<span class='hint'>You apply the coating to the blade.</span>")
```

Comment:
add a message here saying something like "this blade is already coated" if it's full